cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/kafkaassigner/KafkaAssignerDiskUsageDistributionGoal.java

```diff
@@ -599,9 +603,15 @@ public int compareTo(ReplicaWrapper o) {
       if (result != 0) {
         return result;
       } else {
-        if (this.replica() == Replica.MAX_REPLICA || o.replica() == Replica.MIN_REPLICA) {
+        if ((this.replica() == Replica.MAX_REPLICA || this.replica() == Replica.MIN_REPLICA)
+            && (o.replica() == Replica.MAX_REPLICA || o.replica() == Replica.MIN_REPLICA)) {
+          // MIN and MAX replica with the same size are equal.
+          return 0;
+        } else if (this.replica() == Replica.MAX_REPLICA) {
```

Comment:
I think we have bug here.

`this.replica() == Replica.MAX_REPLICA` will return incorrect value here if both `this.replica() == Replica.MAX_REPLICA && o.replica() == Replica.MAX_REPLICA`